### PR TITLE
Convert buffer to slice so user doesn't need `unsafe`

### PR DESCRIPTION
**Pull request by tgross35**
Wasn't sure if there is a reason it isn't like this already, but it's nice to save the step on the user end

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -135,12 +134,13 @@ macro_rules! algorithm {
         }
         #[no_mangle]
         #[link_section = ".entry"]
-        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, data: *const u8) -> u32 {
+        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, buf: *const u8) -> u32 {
```
Comment (Yatekii) [inline src/lib.rs]:
```suggestion
        pub unsafe extern "C" fn ProgramPage(addr: u32, size: u32, data: *const u8) -> u32 {
```

I rather keep this called `data` :)

**Review by Yatekii — COMMENTED** [review]
Thanks! :)
I think it might have been chose to appease to a specific flash interface (rpi2040 is my guess) which requires a pointer. But I think generally a slice is nicer :)

I added a minor nit ;)

**Comment by tgross35** [comment]
> Thanks! :) I think it might have been chose to appease to a specific flash interface (rpi2040 is my guess) which requires a pointer. But I think generally a slice is nicer :)

Ah, makes sense. `.as_ptr()` would work for those cases I suppose

> I added a minor nit ;)

All set!

**Review by Yatekii — APPROVED** [review]
Thank you!

**Comment by tgross35** [comment]
Thanks!

**Pull request merged**